brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/brooklyn/Alien4CloudApplication.java

```diff
@@ -0,0 +1,204 @@
+package io.cloudsoft.tosca.a4c.brooklyn;
+
+import java.util.Map;
+import java.util.Set;
+
+import javax.annotation.Nullable;
+
+import org.apache.brooklyn.api.catalog.CatalogItem;
+import org.apache.brooklyn.api.mgmt.ManagementContext;
+import org.apache.brooklyn.api.policy.Policy;
+import org.apache.brooklyn.core.catalog.internal.CatalogUtils;
+import org.apache.brooklyn.util.collections.MutableMap;
+import org.apache.brooklyn.util.text.Strings;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Predicate;
+import com.google.common.collect.Iterables;
+
+import alien4cloud.model.topology.AbstractPolicy;
+import alien4cloud.model.topology.GenericPolicy;
+import alien4cloud.model.topology.NodeGroup;
+import alien4cloud.model.topology.NodeTemplate;
+import alien4cloud.model.topology.RelationshipTemplate;
+import alien4cloud.model.topology.Topology;
+
+public class Alien4CloudApplication implements ToscaApplication{
+
+    private final String name;
+    private final Topology deploymentTopology;
+    private final String deploymentId;
+
+    public Alien4CloudApplication(String name, Topology deploymentTopology, String deploymentId) {
+        this.name = name;
+        this.deploymentTopology = deploymentTopology;
+        this.deploymentId = deploymentId;
+    }
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public String getDeploymentId() {
+        return deploymentId;
+    }
+
+    @Override
+    public String getId() {
+        return deploymentTopology.getId();
+    }
+
+    @Override
+    public String getDelegateId() {
+        return deploymentTopology.getDelegateId();
+    }
+
+    public Topology getTopology() {
+        return deploymentTopology;
+    }
+
+    private Map<String, NodeTemplate> getNodeTemplates(){
+        return getTopology().getNodeTemplates();
+    }
+
+    public NodeTemplate getNodeTemplate(String id) {
+        return getNodeTemplates().get(id);
+    }
+
+    @Override
+    public Optional<String> getNodeName(String id) {
+        String name = getNodeTemplate(id).getName();
+        if(Strings.isBlank(name)){
+            return Optional.absent();
+        }
+        return Optional.of(name);
+    }
+
+    @Override
+    public Map<String, String> getKeywordMap(String id) {
+        return getKeywordMap(getNodeTemplate(id));
+    }
+
+    public Map<String, String> getKeywordMap(NodeTemplate nodeTemplate) {
+        return MutableMap.of(
```

Comment:
it's to support the possibility of null values.  Could use a builder for ImmutableMap if you prefer?